src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -189,27 +232,28 @@ wait_gmio(xrtDeviceHandle dhdl, const char *gmio_name)
   device->wait_gmio(gmio_name);
 }
 
-static int
-start_profiling(xrtDeviceHandle dhdl, int option, const char *port1_name, const char *port2_name, uint32_t value)
-{
-  auto device = xrt_core::device_int::get_core_device(dhdl);
-  return device->start_profiling(option, port1_name, port2_name, value);
-}
+// C-API Event handles are inserted to this map.
+static std::map<int, std::shared_ptr<xrt::aie::event_impl>> event_cache;
```

Comment:
I am unsure about this cache?  I understand that we need to manage handle for C-APis, but I am not sure what the handle is, seems to come from core_device->start_priofiling,  I have no idea what it represents and why?

Reply:
That's right Soren, this is to manage handle for C APIs. Yes, we get it from start_profiling() and this function will configure the performance counters in the AI Engine and starts profiling. Then it returns a handle to be used by read_profiling and stop_profiling. 
If the specification is incorrect or there is insufficient hardware resources to perform the profiling, an invalid_handle is returned.